pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -68,6 +68,18 @@ func (p Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionIn
 		if err = AddServiceAccountRoles(ctx, projectID, masterSA, GetMasterRoles()); err != nil {
 			return fmt.Errorf("failed to add master roles: %w", err)
 		}
+
+		// Add additional roles for shared VPC
+		if len(in.InstallConfig.Config.Platform.GCP.NetworkProjectID) > 0 {
+			projID := in.InstallConfig.Config.Platform.GCP.NetworkProjectID
+			// Add roles needed for creating firewalls
+			dnsRole := fmt.Sprintf("projects/%s/roles/dns.networks.bindPrivateDNSZone", projID)
```

Comment:
The role "dns.networks.bindPrivateDNSZone" is a custom role, which exists in QE's current GCP host project. What if the role doesn't exist? BTW, in terms of provisioning firewall-rules in the host project, the 2 roles returned by `GetSharedVPCRoles` should be enough already. Thanks!

Reply:
Thanks @jianli-wei, Patrick pointed out this doc https://docs.openshift.com/container-platform/4.16/installing/installing_gcp/installing-gcp-account.html#minimum-required-permissions-ipi-gcp-xpn
where it lists dns.networks.bindPrivateDNSZone in addition to the other roles.

Reply:
We cleared this up in slack but adding here for posterity that @jianli-wei is correct. The permission is needed by the installer, but not by the service account so it can be removed here. We will need to clear this up in docs.